x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/logic/result_settings_logic.ts

```diff
@@ -0,0 +1,200 @@
+      true,
+      {
+        initializeResultFields: () => false,
+      },
+    ],
+    saving: [
+      false,
+      {
+        initializeResultFields: () => false,
+        saving: () => true,
+      },
+    ],
+    openModal: [
+      OpenModal.None,
+      {
+        initializeResultFields: (_, { resultFields }: { resultFields: FieldResultSettingObject }) =>
+          areFieldsAtDefaultSettings(resultFields) ? OpenModal.ConfirmModifyModal : OpenModal.None,
+        closeModals: () => OpenModal.None,
+        resetAllFields: () => OpenModal.None,
+        openConfirmResetModal: () => OpenModal.ConfirmResetModal,
+        openConfirmSaveModal: () => OpenModal.ConfirmSaveModal,
+        saving: () => OpenModal.None,
+      },
+    ],
+    nonTextResultFields: [
+      {},
+      {
+        initializeResultFields: (
+          _,
+          { nonTextResultFields }: { nonTextResultFields: FieldResultSettingObject }
+        ) => nonTextResultFields,
+        clearAllFields: (nonTextResultFields) => clearAllFields(nonTextResultFields),
+        resetAllFields: (nonTextResultFields) => resetAllFields(nonTextResultFields),
+        updateField: (nonTextResultFields, { fieldName, settings }) =>
+          fieldName in nonTextResultFields
+            ? { ...nonTextResultFields, [fieldName]: settings }
+            : nonTextResultFields,
+      },
+    ],
+    textResultFields: [
+      {},
+      {
+        initializeResultFields: (
+          _,
+          { textResultFields }: { textResultFields: FieldResultSettingObject }
+        ) => textResultFields,
+        clearAllFields: (textResultFields) => clearAllFields(textResultFields),
+        resetAllFields: (textResultFields) => resetAllFields(textResultFields),
+        updateField: (textResultFields, { fieldName, settings }) =>
+          fieldName in textResultFields
+            ? { ...textResultFields, [fieldName]: settings }
+            : textResultFields,
+      },
+    ],
+    resultFields: [
+      {},
+      {
+        initializeResultFields: (_, { resultFields }: { resultFields: FieldResultSettingObject }) =>
+          resultFields,
+        clearAllFields: (resultFields) => clearAllFields(resultFields),
+        resetAllFields: (resultFields) => resetAllFields(resultFields),
+        updateField: (resultFields, { fieldName, settings }) =>
+          fieldName in resultFields ? { ...resultFields, [fieldName]: settings } : resultFields,
+      },
+    ],
+    serverResultFields: [
+      {},
+      {
+        initializeResultFields: (
+          _,
+          { serverResultFields }: { serverResultFields: ServerFieldResultSettingObject }
+        ) => serverResultFields,
+        clearAllFields: (serverResultFields) => clearAllServerFields(serverResultFields),
+        resetAllFields: (serverResultFields) => resetAllServerFields(serverResultFields),
+        updateField: (serverResultFields, { fieldName, settings }) => {
+          return fieldName in serverResultFields
```

Comment:
[optional] We might want to consider converting all these `key in object` usages to `obj.hasOwnProperty(key)` as a general best practice when dealing with plain old Javascript objects / not intentionally checking for prototype keys or using ES6 getters/setters 🤷 

In this scenario I can't _super_ see anyone potentially creating a bug with `in` (maybe if they create a key that conflicts with a prototype class, like `constructor`?), but IMO it's just generally a good habit to get into

Reply:
That's a great call, I will definitely update these.

Reply:
https://github.com/elastic/kibana/pull/94629/commits/e60116aea1ec925153d941f81aa2a2a67a103288